pkg/destroy/aws/aws.go

```diff
@@ -760,7 +760,7 @@ func deleteEC2RouteTablesByVPC(client *ec2.EC2, vpc string, logger logrus.FieldL
 		},
 		func(results *ec2.DescribeRouteTablesOutput, lastPage bool) bool {
 			for _, table := range results.RouteTables {
-				lastError := deleteEC2RouteTableObject(client, table, logger.WithField("table", *table.RouteTableId))
+				lastError = deleteEC2RouteTableObject(client, table, logger.WithField("table", *table.RouteTableId))
```

Comment:
@wking I realize that I am late to the party on this, but I don't think that this is the right solution here. This ends up overwriting any previous errors that there may have been. I think we want the result of `delet4eEc2RouteTableObject` to be stored in a variable in the local scope and only stored in `lastError` when there is an actual error.

```
if err := deleteEC2RouteTableObject(...); err != nil {
  lastError = errors.Wrapf(err, ...)
  logger.Info(lastError)
}
```

Reply:
> I think we want the result of delet4eEc2RouteTableObject to be stored in a variable in the local scope and only stored in lastError when there is an actual error.

Ah right.  But let's do that with a different variable name (just `err`?).  For you code comment, I'd rather only log errors at debug level, and then only when we're throwing them out.  So:

```go
if err := deleteEC2RouteTableObject(...); err != nil {
  err = errors.Wrapf(err, ...)
  If lastError != nil {
    logger.Debug(lastError)
  }
  lastError = err
}
```

Thoughts?

Reply:
Yes, only logging the error when it is about to be swallowed by another error makes sense. I also agree that logging it at the debug level fits more consistently with other logged errors.

Reply:
I've filed #1439 addressing this issue.